discovery_engine_core/core/src/engine.rs

```diff
@@ -1429,7 +1429,8 @@ impl Engine {
                 .map_err(Error::InvalidStack)?;
         self.state.reset();
 
-        self.update_stacks_for_all_markets(&[], &[], self.core_config.request_new)
+        self.request_after = 0;
+        self.update_stacks_for_all_markets(&[], &[], usize::MAX)
```

Comment:
this mirrors the behavior when a new engine is created, we want to update the stacks uncoditionally. since we already cleared the stack before, this hasn't caused any issues, but it's still better to have a consistent behavior.

Reply:
I see, so this was previously not quite right and is now corrected. Thanks!